src/App.jsx

```diff
@@ -14,33 +17,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_BY_NAME = 'name';
+const SORT_BY_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function sortingAndReversing(array, sortType, isReverse) {
+  const arrayCopy = [...array];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  switch (sortType) {
+    case SORT_BY_NAME:
+      arrayCopy.sort((goods1, goods2) => goods1.localeCompare(goods2));
+      break;
+    case SORT_BY_LENGTH:
+      arrayCopy.sort((goods1, goods2) => goods1.length - goods2.length);
+      break;
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    default:
+      break;
+  }
+
+  if (isReverse) {
+    arrayCopy.reverse();
+  }
+
+  return arrayCopy;
+}
+
+export function App() {
+  const [sortType, setSortType] = useState('');
+  const [reverseCondition, setReverseCondition] = useState(false);
```

Comment:
Use descriptive variable names to make your code more readable, maintainable, and easier to understand at a glance.

```suggestion
  const [isReversedGoods, setIsReversedGoods] = useState(false);
```